src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -0,0 +1,259 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp"
+
+// BoostMultiArray is used internally in odeint, so BoostMultiArray MUST be
+// included before odeint
+#include "DataStructures/BoostMultiArray.hpp"
+
+#include <boost/numeric/odeint.hpp>
+#include <pup_stl.h>
+
+#include "Domain/FunctionsOfTime/QuaternionHelpers.hpp"
+#include "NumericalAlgorithms/OdeIntegration/OdeIntegration.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace domain::FunctionsOfTime {
+template <size_t MaxDerivReturned>
+QuaternionFunctionOfTime<MaxDerivReturned>::QuaternionFunctionOfTime(
+    double t, std::array<DataVector, 1> initial_func,
+    gsl::not_null<
+        domain::FunctionsOfTime::PiecewisePolynomial<MaxDerivReturned>*>
+        omega_f_of_t_ptr,
+    double expiration_time) noexcept
+    : stored_quaternions_and_times_{{t, std::move(initial_func), false}},
+      omega_f_of_t_ptr_(omega_f_of_t_ptr),
+      expiration_time_(expiration_time) {}
+
+template <size_t MaxDerivReturned>
+std::unique_ptr<FunctionOfTime>
+QuaternionFunctionOfTime<MaxDerivReturned>::get_clone() const noexcept {
+  return std::make_unique<QuaternionFunctionOfTime>(*this);
+}
+
+template <size_t MaxDerivReturned>
+void QuaternionFunctionOfTime<MaxDerivReturned>::pup(PUP::er& p) {
+  FunctionOfTime::pup(p);
+  p | stored_quaternions_and_times_;
+  p | expiration_time_;
+}
```

Comment:
What happens to `omega_f_of_t_ptr_` when this class is pupped?

Reply:
See comment below.